x-pack/solutions/observability/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -265,7 +265,7 @@ function getRootWaterfallTransaction(
   }
 }
 
-export function getLegendsAndColorBy(waterfallItems: IWaterfallItem[]) {
+export function assignColorsAndGenerateLegends(waterfallItems: IWaterfallItem[]) {
```

Comment:
Looks like this is doing multiple things. Should we split it into 2 functions? One to return the `legends` and `colorBy` and another to assign the colors to `waterfallItems`?